### PR TITLE
Mason fixes from #16091 and #16078

### PR DESCRIPTION
**Pull request by ben-albrecht**
This PR fixes some failures caused by recent mason PRs:

- #16091 caused an issue with parsing TOMLs in the mason-registry, breaking `mason search` (this was not visible in testing AFAIK)
- #16078 caused some failures in mason testing

To address issues from #16091, this PR essentially reverts the changes in the TOML module, and futurizes the test it added.

To address issues from #16078, this PR takes the refactor of mason update further and generalizes the `skipUpdate` pattern across all of mason. The `skipUpdate` pattern is: default to `MASON_OFFLINE`, and `--update` or `--no-update` can override the default. The `updateLock` and `updateRegistries` interfaces were also updated along the way. Some tests had to be updated to account for this. 

- [x] `start_test test/mason test/library/packages/UnitTest test/library/packages/TOML`

### CODE REVIEW

**Comment by ben-albrecht** [comment]
@ankingcodes - would appreciate a post-merge review as well.

@vasslitvinov - thanks for offering to review!

File: modules/packages/TOML.chpl [hunk 1]
```diff
@@ -1217,11 +1218,11 @@ module TomlReader {
             ld = "^\\d{4}-\\d{2}-\\d{2}",
             ti = "^\\d{2}:\\d{2}:\\d{2}(.\\d{6,})?";
 
-      const pattern = compile('|'.join(bracketContents,
-                                       brackets,
-                                       tblName,
-                                       doubleQuotes,
+
```
Comment (vasslitvinov) [inline modules/packages/TOML.chpl:1221]: I prefer avoiding double newlines. For this PR either way is fine.

Comment (ben-albrecht) [reply modules/packages/TOML.chpl]: You got it boss

**Comment by ben-albrecht** [comment]
Testing looks good:

```
> start_test test/mason test/library/packages/UnitTest test/library/packages/TOML
...
[Summary: #Successes = 147 | #Failures = 0 | #Futures = 0 | #Warnings = 0 ]
```

**Pull request merged**